src/tufup/client.py

```diff
@@ -184,6 +184,13 @@ def check_for_updates(
                 sys.exit()
             return None
         # check for new target files (archives and patches)
+        # look through all files  in self.current_archive_local_path.parent, and remove all that is not self.current_archive_local_path
+        # this is to save space
+        for file in self.current_archive_local_path.parent.iterdir():
+            metadata_for_file = TargetMeta.parse_filename(file.name)
+            logger.debug(f'checking {metadata_for_file}')
+            if TargetMeta(name=metadata_for_file["name"], version=metadata_for_file["version"], is_archive=metadata_for_file["suffix"]) < self.current_archive:
```

Comment:
`TargetMeta` can be instantiated directly from the file path, so no need for the `parse_filename()` call.

https://github.com/dennisvang/tufup/blob/473045f2f8551e3bef9febd8b9969851c20df49a/src/tufup/common.py#L66

To be safe, I would remove any and all `.patch` files and `.tar.gz` files not equal to `current_archive`.

It would be nice if this could be placed in a separate method, e.g. `clean_cache()`, which is then called from here.

The `clean_cache()` method would need a test as well.